commands/channelmarshaler.go

```diff
@@ -11,16 +12,25 @@ type ChannelMarshaler struct {
 
 func (cr *ChannelMarshaler) Read(p []byte) (int, error) {
 	if cr.reader == nil {
-		val, more := <-cr.Channel
-		if !more {
-			return 0, io.EOF
-		}
+		select {
+		case val, more := <-cr.Channel:
+			if !more {
+				select {
+				case err := <-cr.Errors:
+					return 0, err
+				default:
```

Comment:
@mappum I was hoping these changes would make it print out an error if it encounters one, but it doesnt seem to be doing so.